src/pages/nft-id/questions/quiz.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FunctionComponent, useState, useEffect, useCallback } from 'react';
+import Cookies from 'universal-cookie';
+import { useStore } from 'src/stores';
+import { Question } from 'src/components/nft-id/question';
+import { QUIZ_ATTEMPTS, QUIZ_LOCKED, QUIZ_PASSED } from './consts';
+import GettingStarted from './getting-started';
+import QuizComplete from './quiz-complete';
+
+type UserAnswer = {
+	id: number;
+	answers: number[];
+};
+
+type QuizPageProps = {
+	isLockedOut: boolean;
+	onComplete: () => void;
+};
+
+const cookies = new Cookies();
+
+const QuizPage: FunctionComponent<QuizPageProps> = observer(({ isLockedOut, onComplete }) => {
+	const [currentStep, setCurrentStep] = useState(0);
+	const [userScore, setUserScore] = useState<number>(0);
+	const [userAnswers, setUserAnswers] = useState<UserAnswer[]>([]);
+	const currentQuestion = currentStep - 1;
+	const { questionsStore } = useStore();
+	const quizQuestions = questionsStore.questions;
+
+	const handleGettingStartedClick = useCallback(() => {
+		setCurrentStep(1);
+	}, []);
+
+	const handleQuestionCompletion = useCallback(
+		(answerData: UserAnswer) => {
+			const matchIndex = userAnswers.findIndex(answer => answer.id === answerData.id);
+			matchIndex !== -1
+				? setUserAnswers(previousAnswers => [
+						...previousAnswers.filter(answer => answer.id !== answerData.id),
+						answerData,
+				  ])
+				: setUserAnswers(previousAnswers => [...previousAnswers, answerData]);
+
+			setCurrentStep((prevState: number) => prevState + 1);
+		},
+		[userAnswers]
+	);
+
+	const handlePreviousQuestionClick = useCallback(() => {
+		setCurrentStep((prevState: number) => prevState - 1);
+	}, []);
+
+	const handleQuizFailed = useCallback(() => {
+		const previousAttempts = localStorage.getItem(QUIZ_ATTEMPTS);
+		if (previousAttempts !== null) {
+			const formattedPreviousAttempts = parseInt(previousAttempts);
+
+			if (formattedPreviousAttempts >= 2) {
+				const today = new Date();
+				const tomorrow = new Date();
+				tomorrow.setDate(today.getDate() + 1);
+				cookies.set(QUIZ_LOCKED, 'true', {
+					path: '/',
+					expires: tomorrow,
+				});
+				localStorage.setItem(QUIZ_ATTEMPTS, '0');
+
+				return;
+			}
+
+			localStorage.setItem(QUIZ_ATTEMPTS, JSON.stringify(formattedPreviousAttempts + 1));
+		}
+	}, []);
+
+	const handleQuizPassed = useCallback(() => {
+		cookies.set(QUIZ_PASSED, 'true', {
+			path: '/',
+		});
+		onComplete();
+	}, [onComplete]);
+
+	const fetchQuizResult = useCallback(
+		async (answers: UserAnswer[]) => {
+			const results = await questionsStore.submitAnswers(answers);
+			setUserScore(results.totalScore);
+
+			if (results.totalScore < 80 && currentStep >= 5) {
+				handleQuizFailed();
+			}
+		},
+		[questionsStore, currentStep, handleQuizFailed]
+	);
+
+	useEffect(() => {
+		if (currentStep === 6) {
+			fetchQuizResult(userAnswers);
+		}
+	}, [userAnswers, currentStep, fetchQuizResult]);
+
+	const renderQuestion = () => {
```

Comment:
Instead of doing this function, can we just assign variables?

```
let content = null;
if (currentStep === 0 && !isLockedOut) {
  content = <GettingStarted onClick={handleGettingStartedClick} />;
} else if (...


return (<div>{content}</div>)
```